spinnman/connections/scp_request_pipeline.py

```diff
@@ -154,7 +143,21 @@ def send_request(self, request, callback, error_callback):
         # If all the channels are used, start to receive packets
         while (self._n_channels is not None and
                 self._in_progress >= self._n_channels):
-            self._do_retrieve(self._intermediate_channel_waits, 0.1)
+            self._do_retrieve(
+                self._intermediate_channel_waits, self._packet_timeout)
+
+        # Get the next sequence to be used
+        sequence = self._get_next_sequence_number()
+
+        # Update the packet and store required details
+        request.scp_request_header.sequence = sequence
+        request_data = self._connection.get_scp_data(request)
+        self._requests[sequence] = request
+        self._request_data[sequence] = request_data
+        self._retries[sequence] = self._n_retries
+        self._callbacks[sequence] = callback
+        self._error_callbacks[sequence] = error_callback
+        self._retry_reason[sequence] = list()
```

Comment:
Would it have been better to make `self._retry_reason` be originally a `defaultdict(list)`?